gradle/typescript.gradle

```diff
@@ -0,0 +1,40 @@
+apply plugin: 'com.github.node-gradle.node'
+
+node {
+    download = true
+    distBaseUrl = null
+    version = '16.15.0'
+    workDir = file("${rootProject.projectDir}/.gradle/nodejs")
+}
+
+task lint( type: NpmTask, dependsOn: npmInstall ) {
+    description = 'Lint TS, JS, and JSON'
+    args = [ 'run', 'lint:quiet' ]
+    outputs.upToDateWhen { false }
+}
+
+task typescript( type: NpmTask, dependsOn: [ npmInstall, lint ] ) {
```

Comment:
typescript does not depend on lint
also don't make more tasks than needed.
typescript, copyDefinitions, definitions seem like good candidates for just one npm task

Reply:
Okay. Does `typescript` sound like a good name for all of that?

Reply:
ok with me.